server/src/main/java/com/broadcom/lsp/cobol/service/utils/CustomThreadPoolExecutorService.java

```diff
@@ -0,0 +1,73 @@
+package com.broadcom.lsp.cobol.service.utils;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+
+import java.util.concurrent.*;
+
+/**
+ * {@inheritDoc}
+ *
+ * <p>Implements API for our CustomThreadPoolExecutor created using ThreadPoolExecutor and
+ * ScheduledThreadPoolExecutor.
+ *
+ * <p>The main configuration parameters are: corePoolSize, maximumPoolSize, and keepAliveTime. The
+ * pool consists of a fixed number of core threads that are kept inside all the time, and some
+ * excessive threads that may be spawned and then terminated when they are not needed anymore.
+ */
+@Singleton
+public class CustomThreadPoolExecutorService implements CustomThreadPoolExecutor {
+
+  private int corePoolSize;
+  private int maximumPoolSize;
+  private int keepAliveTime;
+  private int corePoolSizeForScheduledThreadPool;
+
+  @Inject
+  public CustomThreadPoolExecutorService(
+      @Named("CORE-POOL-SIZE") int corePoolSize,
+      @Named("MAX-POOL-SIZE") int maximumPoolSize,
+      @Named("KEEP-ALIVE-TIME-IN-SECONDS") int keepAliveTim,
+      @Named("CORE-POOL-SIZE-FOR-SCHEDULED-POOL") int corePoolSizeForScheduledThreadPool) {
+    this.corePoolSize = corePoolSize;
+    this.maximumPoolSize = maximumPoolSize;
+    this.keepAliveTime = keepAliveTime;
+    this.corePoolSizeForScheduledThreadPool = corePoolSizeForScheduledThreadPool;
+  }
+
+  private ExecutorService getThreadPoolExecutor(
+      int corePoolSize, int maxPoolSize, long keepAliveTime) {
+    return new ThreadPoolExecutor(
```

Comment:
I suppose that this getter also must return the same ThreadPool each time.
I see that you want to rewrite `getScheduledThreadPoolExecutor`, I just want to mark this place for not to forget.

Reply:
> I suppose that this getter also must return the same ThreadPool each time.
> I see that you want to rewrite `getScheduledThreadPoolExecutor`, I just want to mark this place for not to forget.

done